katdal/chunkstore_s3.py

```diff
@@ -62,68 +116,55 @@ class S3ChunkStore(ChunkStore):
 
     Parameters
     ----------
-    client : :class:`botocore.client.S3` object
-        Pre-configured botocore S3 client
+    session : :class:`requests.Session` object
+        Pre-configured session
+    url : str
+        Base URL for the S3 service
 
     Raises
     ------
     ImportError
-        If botocore is not installed (it's an optional dependency otherwise)
+        If requests is not installed (it's an optional dependency otherwise)
     """
 
-    def __init__(self, client):
-        if not botocore:
-            raise _botocore_import_error
-        error_map = {EndpointConnectionError: StoreUnavailable,
-                     ConnectionError: StoreUnavailable,
-                     requests.exceptions.RequestException: StoreUnavailable,
-                     client.exceptions.NoSuchKey: ChunkNotFound,
-                     client.exceptions.NoSuchBucket: ChunkNotFound}
+    def __init__(self, session_factory, url):
+        if not requests:
+            raise _requests_import_error
+        try:
+            # Quick smoke test to see if the S3 server is available,
+            # by listing buckets
+            with session_factory() as session:
+                with contextlib.closing(session.get(url)) as response:
+                    _raise_for_status(response)
```

Comment:
Could a 404 happen at this stage? That should map to `StoreUnavailable` here instead of `ChunkNotFound`, I reckon.

Reply:
This doesn't involve `_standard_errors`, so the StoreUnavailable two lines below will trigger.